covsirphy/phase/trend.py

```diff
@@ -36,6 +36,9 @@ def __init__(self, sr_df):
             raise ValueError("The length of @sr_df must be over 2.")
         # Setting for analysis
         self.result_df = None
+        self.max_linear_fit_err = 0.5
```

Comment:
Please create `Trend._run_linear()` and `Trend._run_negative_exponential()`. Then, add `func` argument to `Trend.run()`.

Reply:
Why create these two functions? Isn't enough to add only in `__init__` of the `Trend` class a variable `self.fit_fnc` and set this to either `self.linear` or `self.negative_exp` depending on the `Trend.run(func)`?

Reply:
Thank you, your solution (`.run(func)`) is better.